### DISCUSSION TITLE
Getting timing path using the GUI

### DISCUSSION THREAD

**Discussion by xobs** [The-OpenROAD-Project/OpenROAD#3147] [Q&A] [answered]
I saw your presentation at Latchup, and was really impressed with how you were able to dive into the design and figure out things such as the critical path from the GUI.

I'm using the OpenLane flow, and have the odb file open:

![image](https://user-images.githubusercontent.com/238325/230816281-25f560de-e476-45ba-b3ec-5f6b595fae93.png)

However, it is not clear how to get timing information to show up. I am viewing the newest and largest odb file from `openlane/mimi-bare/runs/mimi-bare/tmp` so I assume it's the file with all of the relevant data.

Is there some setting I need to change in openlane in order to find the relevant data? Or is there another flow I should be using?

**Comment by maliberty** [comment]
You need to hit the update button.  It takes a while so we don't do it automatically in case you want to do any setting first.

**Reply by xobs** [reply]
For some reason, in this design the update happens instantly in both the Timing Report and the Clock Tree Viewer, but both windows remain empty.

**Reply by QuantamHD** [reply, marked as answer]
Yeah so an odb file isn't enough to actually get the timing report. You need to read in the liberty files and sdc file after running read_db.

**Reply by xobs** [reply]
Thanks, the `sdc` was the missing piece.

With this script, I get clock support in openroad:

```
read_db routing/16-global.odb
read_liberty /si/conda/share/pdk/gf180mcuC/libs.ref/gf180mcu_fd_sc_mcu7t5v0/liberty/gf180mcu_fd_sc_mcu7t5v0__tt_025C_5v00.lib
read_sdc 12-minimax.sdc
```

**Reply by maliberty** [reply]
You may also need to load parasitics (eg spef) to get accurate timing at the end of the flow.  If you are at global routing you may want to run ```estimate_parasitics -global_routing```

**Reply by xobs** [reply]
![image](https://user-images.githubusercontent.com/238325/230823004-97517621-8135-472a-8860-7762c985cc70.png)

Thanks for all of the help -- this is an amazing superpower
